bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -774,7 +774,7 @@ impl Client {
                 }
             });
 
-            *self.delegate.write().unwrap() = Some(Arc::from(delegate));
+            self.delegate.get_or_init(|| Arc::from(delegate));
```

Comment:
Same: let's throw if the delegate was already set, and let's add a doc comment explaining this invariant that it's a no op if the delegate was already set.

Probably you need to do the check at the start of the function, to avoid spawning a task that will be detached forever if the delegate was already set.

(Overall, I would have preferred this to be in a separate PR, because it's unrelated to the current one.)